rust/src/tx_builder.rs

```diff
@@ -304,18 +305,17 @@ impl TransactionBuilder {
                 bootstraps: BTreeSet::new(),
             },
             validity_start_interval: None,
+            mint: None
         }
     }
 
     /// does not include refunds or withdrawals
     pub fn get_explicit_input(&self) -> Result<Coin, JsError> {
-        self
-            .inputs
+        self.inputs
             .iter()
-            .try_fold(
-                to_bignum(0),
-                |acc, ref tx_builder_input| acc.checked_add(&tx_builder_input.amount)
-            )
+            .try_fold(to_bignum(0), |acc, ref tx_builder_input| {
+                acc.checked_add(&tx_builder_input.amount.coin())
+            })
     }
     /// withdrawals and refunds
     pub fn get_implicit_input(&self) -> Result<Coin, JsError> {
```

Comment:
Probably `get_explicit_input`, `get_implicit_input` and `get_explicit_output` should return `Value` instead of `Coin`

Reply:
Hmmm, I guess I can add `checked_add` to `Value`, maybe `Eq` and `PartialOrd` too (and expose them to js?). 
With `checked_sub` the code for the `add_change_if_needed` would look a bit simpler (or at least, the same as it was before with only scalar values), so that's worth adding too.